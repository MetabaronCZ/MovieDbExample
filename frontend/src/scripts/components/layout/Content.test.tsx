import { describe, expect, it } from '@jest/globals';
import { getByRole, getByText } from '@testing-library/dom';

import { Content } from './Content';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/layout/Content', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(<Content>CONTENT</Content>, {
      wrapper: TestComponentWrapper,
    });
    expect(container).toMatchSnapshot();

    const content = getByText(container, 'CONTENT');
    expect(content).toBeInTheDocument();

    const main = getByRole(container, 'main');
    expect(main).toBeInTheDocument();
  });
});
