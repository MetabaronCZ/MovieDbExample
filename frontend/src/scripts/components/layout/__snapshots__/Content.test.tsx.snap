// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`components/layout/Content should render properly 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  flex: 1;
  gap: 16px;
}

.c1 {
  width: 100%;
  min-width: 0;
  padding: 16px 24px;
  overflow-y: auto;
}

<div>
  <main
    class="c0 c1"
  >
    CONTENT
  </main>
</div>
`;
